### PR TITLE
profiling: support authentication of the profiling api 

### PR DESCRIPTION
**Pull request by rleungx**
This PR supports the authentication of the profiling API.

### CODE REVIEW

**Comment by breeswish** [comment]
This will break the download functionality. I think you can wait @Deardrops to give a solution for it.

**Comment by breeswish** [comment]
You can update your PR according to https://github.com/pingcap-incubator/tidb-dashboard/pull/165 now :)

File: ui/src/apps/logSearching/components/SearchProgress.tsx [hunk 1]
```diff
@@ -207,14 +200,14 @@ export default function SearchProgress({
...
-    
+
     const res = await client.dashboard.logsDownloadAcquireTokenGet(keys)
     const token = res.data
     if (!token) {
       return
     }
     const url = `${DASHBOARD_API_URL}/logs/download?token=${token}`
-    downloadFile(url)
+    window.open(url)
```
Comment (Deardrops) [inline ui/src/apps/logSearching/components/SearchProgress.tsx]: Open a new window may be intercepted by some browser, I suggest download file by creating a hidden `<a>` element and click it.

Comment (breeswish) [reply ui/src/apps/logSearching/components/SearchProgress.tsx]: If opening a new window is intercepted, how will this method work?

Comment (breeswish) [reply ui/src/apps/logSearching/components/SearchProgress.tsx]: How about this solution? https://stackoverflow.com/questions/19026162/javascript-window-open-from-callback Looks like this should be the correct answer.

Comment (Deardrops) [reply ui/src/apps/logSearching/components/SearchProgress.tsx]: OK for me.

**Pull request merged**